### PR TITLE
feat: add fastify adapter

### PR DESCRIPTION
**Pull request by mcansh**
still need to write ~~some tests and~~ an example/template, but using this puppy on a side project: https://shredder.mcan.sh

Signed-off-by: Logan McAnsh <logan@mcan.sh>

### CODE REVIEW

**Comment by mcansh** [comment]
> Thank you for signing the Contributor License Agreement. Let's get this merged! 🥳

hm, thought i fixed sending a comment to an open PR if they've signed already...

File: packages/remix-fastify/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/remix-run/remix",
+    "directory": "packages/remix-fastify"
+  },
+  "bugs": {
+    "url": "https://github.com/remix-run/remix/issues"
+  },
+  "dependencies": {
+    "@remix-run/node": "1.0.4"
```
Comment (MichaelDeBoey) [inline packages/remix-fastify/package.json:14]: Shouldn't this package also have `@remix-run/server-runtime` as dependency? 🤔

Comment (mcansh) [reply packages/remix-fastify/package.json:14]: `@remix-run/node` has `@remix-run/server-runtime` as a dependency :)

Comment (MichaelDeBoey) [reply packages/remix-fastify/package.json:14]: I think all of them should have `@remix-run/server-runtime` as a dependency then or at least a `peerDependency`

File: packages/remix-fastify/__tests__/server-test.ts [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        method: "GET",
+        url: "/"
+      });
+
+      expect(response.statusCode).toBe(200);
+    });
+
+    it("handles status codes", async () => {
+      mockedCreateRequestHandler.mockImplementation(() => async () => {
+        return new Response("", { status: 204 });
```
Comment (MichaelDeBoey) [inline packages/remix-fastify/__tests__/server-test.ts:79]:
```suggestion
        return new Response(null, { status: 204 });
```

See #847

File: packages/remix-fastify/__tests__/server-test.ts [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        );
+        headers.append(
+          "Set-Cookie",
+          "second=two; MaxAge=1209600; Path=/; HttpOnly; Secure; SameSite=Lax"
+        );
+        headers.append(
+          "Set-Cookie",
+          "third=three; Expires=Wed, 21 Oct 2015 07:28:00 GMT; Path=/; HttpOnly; Secure; SameSite=Lax"
+        );
+        return new Response("", { headers });
```
Comment (MichaelDeBoey) [inline packages/remix-fastify/__tests__/server-test.ts:106]:
```suggestion
        return new Response(null, { headers });
```

See #847

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Comment by ryanflorence** [comment]
This is awesome.

But we've got to figure out a better way to support a bunch of adapters without all of them being part of the remix source code. Let's figure this out after the break. I made a linear issue.

**Comment by mcansh** [comment]
yeah i felt the same when i opened it haha 👍

**Comment by rsbh** [comment]
What is the status of fastify integration?
I am fine in using custom solution but i prefer official support.
Can you folks share the status or timeline regarding fastify support

**Comment by chaance** [comment]
@rsbh This isn't a huge priority for us at the moment, sorry! Official support means for more adapters is out of scope at the moment, but if we revisit this in the future we can revisit this discussion.

**Pull request closed without merging**